src/core/DataArray.ts

```diff
@@ -110,7 +110,13 @@ export default function DataArray(props: DataArrayProps) {
       const typedArrayClass = TypedArrayLookup[TYPED_ARRAYS[type]];
       array.setData(toTypedArray(values, typedArrayClass), numberOfComponents);
       if (range) {
+        // for a dataArray of single component, range is cached with both:
+        // index = 0, and
+        // index = numberOfComponents = 1
         array.setRange(range, numberOfComponents);
+        if (numberOfComponents === 1) {
+          array.setRange(range, 0);
+        }
```

Comment:
@floryst I suppose rather than setting the range twice (based on the vtk-js getRange fix), I can just do this:
```js
        if (numberOfComponents === 1) {
          array.setRange(range, 0);
       } else {
          array.setRange(range, numberOfComponents);
        }
```